core/src/main/scala/kafka/log/Log.scala

```diff
@@ -167,18 +172,18 @@ class Log(@volatile var dir: File,
       val filename = file.getName
       if(filename.endsWith(DeletedFileSuffix) || filename.endsWith(CleanedFileSuffix)) {
         // if the file ends in .deleted or .cleaned, delete it
-        file.delete()
+        Files.deleteIfExists(file.toPath)
```

Comment:
My gut feeling would be to keep it simple for now, and have more elaborate error handling later if we find scenarios where it would have helped.  I think it's rare to find a filesystem that has undeletable files that is still usable in practice (but maybe there is a scenario I haven't considered?).